### PR TITLE
Naive first pass at weight decay change

### PR DESCRIPTION
**Pull request by wmcnicho**
@vladfi1 

Very naive first pass at the weight decay change. I doubt this actually works but it does run without exploding.

Opening against `save_to_s3` instead of `main` for a cleaner diff.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    // Use IntelliSense to learn about possible attributes.
+    // Hover to view descriptions of existing attributes.
+    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
+    "version": "0.2.0",
```
Comment (wmcnicho) [inline .vscode/launch.json:5]: I can add this to the `.gitignore` if you want to keep the repo IDE agnostic

File: learner.py [hunk 2]
```diff
@@ -51,4 +54,10 @@ def step(self, batch, initial_states, train=True):
       assert set(watched_names) == set(trainable_names)
       grads = tape.gradient(mean_loss, params)
       self.optimizer.apply(grads, params)
+      # Multiply all weights by decay_rate
+      print("Pre decay params: %s" % (params[0]))
+      for param in params:
+        new_val = tf.math.scalar_mul(self.decay_rate, param)
```
Comment (vladfi1) [inline learner.py]: You can just do `param * self.decay_rate`, no need for scalar_mul.

File: train.py [hunk 3]
```diff
@@ -84,7 +84,8 @@ def main(dataset, expt_dir, num_epochs, epoch_time, save_interval, _config, _log
       _config['learner']['learning_rate'], name='learning_rate')
   learner = Learner(
       learning_rate=learning_rate,
-      policy=policy)
+      policy=policy,
+      decay_rate=0.999)
```
Comment (vladfi1) [inline train.py]: You can make the decay a configurable parameter (like learning_rate).

File: .vscode/launch.json (hunk 1 above)
Comment (vladfi1) [reply .vscode/launch.json:5]: I'm fine with it. Could help any other vscode users out there (like me!).

**Review by vladfi1 — CHANGES_REQUESTED** [review]
Looks good, just a couple things to fix up.

**Comment by wmcnicho** [comment]
Closing in favor of https://github.com/vladfi1/slippi-ai/pull/23
